src/wp-admin/network/site-users.php

```diff
@@ -139,13 +139,27 @@
 
 		case 'promote':
 			check_admin_referer( 'bulk-users' );
+
+			if ( ! current_user_can( 'promote_users' ) ) {
+				wp_die( __( 'Sorry, you are not allowed to edit this user.' ), 403 );
+			}
+
 			$editable_roles = get_editable_roles();
 			$role           = $_REQUEST['new_role'];
 
-			if ( empty( $editable_roles[ $role ] ) ) {
+			// Mocking the `none` role so we are able to save it to the database
+			$editable_roles['none'] = array(
+				'name' => __( '&mdash; No role for this site &mdash;' ),
+			);
+
+			if ( ! $role || empty( $editable_roles[ $role ] ) ) {
```

Comment:
```suggestion
			if ( empty( $editable_roles[ $role ] ) ) {
```

I think we can keep the `empty()` check the same here as we're now mocking `none` as a valid key.

Reply:
```
// Mocking the `none` role so we are able to save it to the database
```

This was lifted straight from single-site `users.php`, so let's update that with this new comment at the same time this lands.